nova/app.py

```diff
@@ -31,9 +31,33 @@
     html.Div(id='output-image'),
 ])
 
+def filter_extracted_text(extracted_text):
+    # Handle text recognition common errors
+    extracted_text = extracted_text.replace("p", "")
+    extracted_text = extracted_text.replace("u", "")
+    extracted_text = extracted_text.replace("m", "")
+    extracted_text = extracted_text.replace("o", "0")
+    extracted_text = extracted_text.replace("O", "0")
+    extracted_text = extracted_text.replace("+0", "+0.")
+    extracted_text = extracted_text.replace("+0..", "+0.")
+    extracted_text = extracted_text.replace("-0", "-0.")
+    extracted_text = extracted_text.replace("-0..", "-0.")
+    extracted_text = extracted_text.replace("+1", "+1.")
+    extracted_text = extracted_text.replace("+1..", "+1.")
+    extracted_text = extracted_text.replace("-1", "-1.")
+    extracted_text = extracted_text.replace("-1..", "-1.")
+    extracted_text = extracted_text.replace("+2", "+2.")
+    extracted_text = extracted_text.replace("+2..", "+2.")
+    extracted_text = extracted_text.replace("-2", "-2.")
+    extracted_text = extracted_text.replace("-2..", "-2.")
+
+    # Hande the occuring of additional . error
+    if len(extracted_text) > 5:
+        extracted_text = extracted_text[:-1]
+    
+    return extracted_text
+
 def extract_data(image):
-    content_type, content_string = image.split(',')
-    pil_image = Image.open(io.BytesIO(base64.b64decode(content_string)))
```

Comment:
This was extracted on the "upper level of execution" to make the function more testable.

Writing tests help you make code more modular as can be seen in this example